ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```
@@ -541,6 +555,83 @@ translateLedgerViewByronToShelleyWrapper =
         maxFor :: SlotNo
         maxFor = addSlots swindow (boundSlot bound)
 
+{-------------------------------------------------------------------------------
+  Translation from one Shelley-based era to another Shelley-based era
+-------------------------------------------------------------------------------}
+
+instance ( ShelleyBasedEra era
+         , ShelleyBasedEra (SL.PreviousEra era)
+         , EraCrypto (SL.PreviousEra era) ~ EraCrypto era
+         ) => SL.TranslateEra era ShelleyTip where
+  translateEra _ (ShelleyTip sno bno (ShelleyHash hash)) =
+      return $ ShelleyTip sno bno (ShelleyHash hash)
+
+instance ( ShelleyBasedEra era
+         , SL.TranslateEra era ShelleyTip
+         , SL.TranslateEra era SL.NewEpochState
+         , SL.TranslationError era SL.NewEpochState ~ Void
+         ) => SL.TranslateEra era (LedgerState :.: ShelleyBlock) where
+  translateEra ctxt (Comp (ShelleyLedgerState tip state _transition)) = do
+      tip'   <- mapM (SL.translateEra ctxt) tip
+      state' <- SL.translateEra ctxt state
+      return $ Comp $ ShelleyLedgerState {
+          shelleyLedgerTip        = tip'
+        , shelleyLedgerState      = state'
+        , shelleyLedgerTransition = ShelleyTransitionInfo 0
+        }
+
+instance ( ShelleyBasedEra era
+         , SL.TranslateEra era SL.Tx
+         , SL.TranslationError era SL.Tx ~ ()
+         ) => SL.TranslateEra era (GenTx :.: ShelleyBlock) where
+  type TranslationError era (GenTx :.: ShelleyBlock) = ()
```

Just forward it and ignore the error in the injection.